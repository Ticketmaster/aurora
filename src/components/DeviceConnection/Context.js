import React from "react";

export const { Provider, Consumer } = React.createContext({ "4g": true });
