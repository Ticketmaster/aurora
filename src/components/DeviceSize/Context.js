import React from "react";

export const { Provider, Consumer } = React.createContext({ isSmall: true });

export const withDeviceSize = Component => props => (
  <Consumer>{value => <Component deviceSize={value} {...props} />}</Consumer>
);
