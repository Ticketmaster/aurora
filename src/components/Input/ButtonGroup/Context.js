import React from "react";

export const { Provider, Consumer } = React.createContext({
  selected: null,
  onChange: null
});

export const withButtonGroupConsumer = Component => props => (
  <Consumer>
    {({ onChange: onGroupChange, value: groupValue = {} }) => (
      <Component
        onGroupChange={onGroupChange}
        groupValue={groupValue}
        {...props}
      />
    )}
  </Consumer>
);
