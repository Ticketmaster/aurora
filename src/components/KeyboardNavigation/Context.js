import React from "react";

export const { Provider, Consumer } = React.createContext({
  focused: null,
  focusSelected: () => null
});
