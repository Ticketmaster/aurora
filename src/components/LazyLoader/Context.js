import { createContext } from "react";

import { PLACEHOLDER_IMAGE } from "./constants";

const { Provider, Consumer } = createContext({
  src: PLACEHOLDER_IMAGE,
  style: null,
  imageRef: null,
  load: () => {}
});

export { Provider, Consumer as LazyLoaderConsumer };
