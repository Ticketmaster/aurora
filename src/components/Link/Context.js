import React from "react";

const { Provider, Consumer } = React.createContext();

export { Provider as LinkListProvider };
export { Consumer as LinkListConsumer };
