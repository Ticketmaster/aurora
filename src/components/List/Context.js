import React from "react";

const { Provider, Consumer } = React.createContext();

export { Provider as ContainerProvider };
export { Consumer as ItemContainerConsumer };
