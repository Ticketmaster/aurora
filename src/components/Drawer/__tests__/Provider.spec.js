/* global window: true */
import React from "react";
import { render, Simulate } from "react-testing-library";

import DrawerProvider from "../Provider";
import { Consumer } from "../Context";

describe("DrawerProvider", () => {
  beforeEach(() => {
    window.scrollTo = jest.fn();
    const modalRoot = global.document.createElement("div");
    modalRoot.setAttribute("id", "modal-root");
    const body = global.document.querySelector("body");
    body.appendChild(modalRoot);
  });

  afterEach(() => {
    const body = global.document.querySelector("body");
    const modalRoot = global.document.getElementById("modal-root");
    body.removeChild(modalRoot);
  });
  it("renders child elements", () => {
    const { container } = render(
      <DrawerProvider>
        <div>Content</div>
      </DrawerProvider>
    );
    expect(container.innerHTML).toMatchSnapshot();
  });

  it("sets content", () => {
    const { getByTestId, container } = render(
      <DrawerProvider>
        <div>
          <Consumer>
            {({ setContent }) => (
              <button
                data-testid="button"
                onClick={() => {
                  setContent(
                    <div data-testid="drawer-content">Drawer Content</div>
                  );
                }}
              >
                ClickMe!
              </button>
            )}
          </Consumer>
          Content
        </div>
      </DrawerProvider>
    );

    Simulate.click(getByTestId("button"));

    expect(container).toMatchSnapshot();
  });

  it("toggles drawer", () => {
    const { getByTestId, container } = render(
      <DrawerProvider>
        <div>
          <Consumer>
            {({ setContent, toggleDrawer }) => (
              <button
                data-testid="button"
                onClick={() => {
                  setContent(
                    <div data-testid="drawer-content">Drawer Content</div>
                  );
                  toggleDrawer();
                }}
              >
                ClickMe!
              </button>
            )}
          </Consumer>
          Content
        </div>
      </DrawerProvider>
    );

    Simulate.click(getByTestId("button"));

    expect(container).toMatchSnapshot();
  });
});
