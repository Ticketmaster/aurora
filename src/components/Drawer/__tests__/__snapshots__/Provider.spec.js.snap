// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerProvider renders child elements 1`] = `"<div class=\\"sc-ifAKCX eaZCGx\\"><div>Content</div></div>"`;

exports[`DrawerProvider sets content 1`] = `
<div>
  <div
    class="sc-ifAKCX eaZCGx"
  >
    <div>
      <button
        data-testid="button"
      >
        ClickMe!
      </button>
      Content
    </div>
  </div>
</div>
`;

exports[`DrawerProvider toggles drawer 1`] = `
<div>
  <div
    class="drawer--open sc-ifAKCX eaZCGx"
  >
    <div>
      <button
        data-testid="button"
      >
        ClickMe!
      </button>
      Content
    </div>
  </div>
</div>
`;
